src/tlo/methods/cervical_cancer.py

```diff
@@ -0,0 +1,1542 @@
+            )
+
+        random_value = random.random()
+
+        if (random_value <= p['prob_cure_stage1'] and df.at[person_id, "ce_hpv_cc_status" == "stage1"]
+            and df.at[person_id, "ce_date_treatment"] == self.sim.date):
+            df.at[person_id, "ce_hpv_cc_status"] = 'none'
+            df.at[person_id, 'ce_current_cc_diagnosed'] = False
+        else:
+            df.at[person_id, "ce_hpv_cc_status"] = 'stage1'
+
+        if (random_value <= p['prob_cure_stage2a'] and df.at[person_id, "ce_hpv_cc_status" == "stage2a"]
+            and df.at[person_id, "ce_date_treatment"] == self.sim.date):
+            df.at[person_id, "ce_hpv_cc_status"] = 'none'
+            df.at[person_id, 'ce_current_cc_diagnosed'] = False
+        else:
+            df.at[person_id, "ce_hpv_cc_status"] = 'stage2a'
+
+        if (random_value <= p['prob_cure_stage2b'] and df.at[person_id, "ce_hpv_cc_status" == "stage2b"]
+            and df.at[person_id, "ce_date_treatment"] == self.sim.date):
+            df.at[person_id, "ce_hpv_cc_status"] = 'none'
+            df.at[person_id, 'ce_current_cc_diagnosed'] = False
+        else:
+            df.at[person_id, "ce_hpv_cc_status"] = 'stage2b'
+
+        if (random_value <= p['prob_cure_stage3'] and df.at[person_id, "ce_hpv_cc_status" == "stage3"]
+            and df.at[person_id, "ce_date_treatment"] == self.sim.date):
+            df.at[person_id, "ce_hpv_cc_status"] = 'none'
+            df.at[person_id, 'ce_current_cc_diagnosed'] = False
+        else:
+            df.at[person_id, "ce_hpv_cc_status"] = 'stage3'
+
+        # Schedule a post-treatment check for 3 months:
+        hs.schedule_hsi_event(
+            hsi_event=HSI_CervicalCancer_PostTreatmentCheck(
+                module=self.module,
+                person_id=person_id,
+            ),
+            topen=self.sim.date + DateOffset(months=3),
+            tclose=None,
+            priority=0
+        )
+
+class HSI_CervicalCancer_PostTreatmentCheck(HSI_Event, IndividualScopeEventMixin):
+    """
+    This event is scheduled by HSI_CervicalCancer_StartTreatment and itself.
+    It is only for those who have undergone treatment for cervical Cancer.
+    If the person has developed cancer to stage4, the patient is initiated on palliative care; otherwise a further
+    appointment is scheduled for one year.
+    """
+
+    def __init__(self, module, person_id):
+        super().__init__(module, person_id=person_id)
+
+        self.TREATMENT_ID = "CervicalCancer_PostTreatmentCheck"
+        self.EXPECTED_APPT_FOOTPRINT = self.make_appt_footprint({"Over5OPD": 1})
+        self.ACCEPTED_FACILITY_LEVEL = '3'
+
+    def apply(self, person_id, squeeze_factor):
+        df = self.sim.population.props
+        hs = self.sim.modules["HealthSystem"]
+
+        assert not pd.isnull(df.at[person_id, "ce_date_diagnosis"])
+        assert not pd.isnull(df.at[person_id, "ce_date_treatment"])
+
+        days_threshold_365 = 365
```

Comment:
could use pd.Dateoffset(years =...) instead of the number of days for ease for reading/comprehension